AdminUi/apps/admin_ui/lib/components/identity/identities_data_table_source.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:admin_api_sdk/admin_api_sdk.dart';
+import 'package:admin_api_types/admin_api_types.dart';
+import 'package:flutter/material.dart';
+
+class IdentityDataTableSource extends DataTableSource {
+  List<IdentityOverview> data = [];
+  Pagination? pagination;
+  int sortColumnIndex = 0;
+  bool sortAscending = true;
+
+  void setData(
+    List<IdentityOverview> identities,
+    Pagination pagination,
+    int columnIndex, {
+    required bool columnAscending,
+  }) {
+    data = identities;
+    this.pagination = pagination;
+    notifyListeners();
+
+    sortColumnIndex = columnIndex;
+    sortAscending = columnAscending;
+    sort();
+  }
+
+  void sort() {
+    data.sort((a, b) {
+      final aValue = _getDisplayValue(a);
+      final bValue = _getDisplayValue(b);
+      return sortAscending ? Comparable.compare(aValue, bValue) : Comparable.compare(bValue, aValue);
+    });
+    notifyListeners();
+  }
+
+  Comparable<Object> _getDisplayValue(IdentityOverview identity) {
+    switch (sortColumnIndex) {
+      case 0:
+        return identity.address;
+      case 1:
+        return identity.tier.name;
+      case 2:
+        return identity.createdWithClient;
+      case 3:
+        return identity.numberOfDevices;
+      case 4:
+        return identity.createdAt;
+      case 5:
+        return identity.lastLoginAt.toString();
+      case 6:
+        return identity.datawalletVersion.toString();
+      case 7:
+        return identity.identityVersion;
+      default:
+        throw Exception('Invalid column index');
+    }
```

Comment:
please use a switch expression here: https://dart.dev/language/patterns#switch-statements-and-expressions

Reply:
Changes made.